core/runtime/src/main/java/io/quarkus/runtime/configuration/LocaleConverter.java

```diff
@@ -29,6 +32,11 @@ public Locale convert(final String value) {
             return null;
         }
 
+        if (localeValue.equals("-")) {
```

Comment:
The idea here is to return `null` only when no locale value is set explicitly, but instead the defaults are both empty. This allows the user explicitly set locale strings like `en-` or `en-US` even when there are no default values.
If the default values are empty and the user doesn't specify any locales, then a higher up layer will end up throwing an exception like:

```
java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.locales not found
```

Reply:
In this case the warning message will at least give the user the opportunity to know to set `user.language` instead of having to set `quarkus.locales` and `quarkus.default-locale`

Reply:
After talking we @gsmet we decided to use `en` as the last line of defence, so this stuff is no longer needed